polybench/2DConvolution.cpp

```diff
@@ -10,6 +10,8 @@
 
 using DATA_TYPE = float;
 
+class conv2D;

```

Comment:
Is there a guide when ComputeCpp requires this? From the spec it's not clear to me why we cannot just `parallel_for<class conv2D>()` here.

Reply:
I think their diagnostic might be a bit overzealous? In any case, for this and `3DConvolution` I ran into a weird issue were I had to both forward-declare it at the top (to avoid the warning), but then also use `class`  inside the template arguments for `parallel_for` because I would otherwise get a `no matching member function call` error.

Reply:
Jesus... Alright, let's just accept that ComputeCpp is... special ;)